pkg/dashboard/resource/project.go

```diff
@@ -237,6 +257,38 @@ func (pr *projectResource) projectToAttributes(project platform.Project) restful
 	return attributes
 }
 
+func (pr *projectResource) validateDisplayNameExclusiveness(request *http.Request, projectInfo *projectInfo) error {
+
+	projectNameSpace := projectInfo.Meta.Namespace
+	if projectInfo.Meta.Namespace == "" {
+		projectNameSpace = pr.getNamespaceFromRequest(request)
+	}
+
+	getProjectsOptions := &platform.GetProjectsOptions{
+		Meta: platform.ProjectMeta{
+			Namespace: projectNameSpace,
+		},
+	}
+
+	nameSpaceProjects, err := pr.getPlatform().GetProjects(getProjectsOptions)
+	if err != nil {
+		return nuclio.WrapErrInternalServerError(errors.Wrap(err, "Failed to get projects"))
+	}
+
+	for _, project := range nameSpaceProjects {
+		if project.GetConfig().Meta.Name == projectInfo.Meta.Name {
+
+			// skip the current project. (relevant when updating, and this project already exists)
+			continue
+		}
+		if project.GetConfig().Spec.DisplayName == projectInfo.Spec.DisplayName {
+			return nuclio.WrapErrConflict(errors.New("Cannot create two projects with the same display name"))
```

Comment:
change the error to work for both update and create.
`Project display name %s already exists for another project`